### PR TITLE
Deprecate methods using Guava ListenableFuture [run-systemtest]

### PR DESCRIPTION
**Pull request by bjorncs**
- com.yahoo.processing.Response.recursiveComplete()
- com.yahoo.processing.response.DataList.complete()
- com.yahoo.processing.response.IncomingData.completed()

Also fixes bug in FutureResponse where `get()` does not return value produced by async task.

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: container-core/src/main/java/com/yahoo/processing/response/DataList.java [hunk 1]
```diff
@@ -73,6 +72,10 @@
      * Making this call on a list which does not support future data always returns immediately and
      * causes no memory synchronization cost.
      */
+    CompletableFuture<DataList<DATATYPE>> future();
```
Comment (bratseth) [inline container-core/src/main/java/com/yahoo/processing/response/DataList.java:75]: Isn't "future" here (and the other places) a bit meaningless? It's a future where ...?

Comment (bratseth) [reply container-core/src/main/java/com/yahoo/processing/response/DataList.java:75]: How about completed()? (Or completeFuture())

**Review by bratseth — COMMENTED** [review]
Great!

File: container-core/src/main/java/com/yahoo/processing/response/DataList.java (hunk 1 above)
Comment (bjorncs) [reply container-core/src/main/java/com/yahoo/processing/response/DataList.java:75]: Good point - I agree. Your second proposal matches well with the existing method. I'll rename.

**Comment by bjorncs** [comment]
@bratseth PTAL

**Review by bratseth — APPROVED** [review]
👍

**Pull request merged**